Add use description of CookieEncryptor and CookieSigner in cookie guide (#104)
yiisoft/docs: guide/en/runtime/cookies.md

After forming a cookie call `addToResponse()` passing an instance of `\Psr\Http\Message\ResponseInterface` to add
corresponding HTTP headers to it.

## Signing cookies

To prevent the substitution of the cookie value, use `Yiisoft\Cookies\CookieSigner`.

> It signs each cookie with a unique prefix hash based on the value of the cookie.

Reviewer: ```suggestion
It signs each cookie with a unique prefix hash based on the value of the cookie and a secret key.
```

```php
$cookie = new \Yiisoft\Cookies\Cookie('identity', 'identityValue');

// The secret key used to sign and validate cookies.
$key = '0my1xVkjCJnD_q1yr6lUxcAdpDlTMwiU';
$signer = new \Yiisoft\Cookies\CookieSigner($key);

$signedCookie = $signer->sign($cookie);
```

To validate and get back the cookie with clean value, use the `validate()` method.

```php
$cookie = $signer->validate($signedCookie);
```

> If the cookie value is tampered or has not been signed before, an `\RuntimeException` will be thrown.

Reviewer: ```suggestion
If the cookie value is tampered with or has not been signed before, a `\RuntimeException` will be thrown.
```

Therefore, if you are not sure that the cookie value was signed earlier, then first use the `isSigned()` method.

```php
if ($signer->isSigned($cookie)) {
    $cookie = $signer->validate($cookie);
}
```

> It makes sense to sign the value of a cookie if you store some important data in it,
> such as identification data, since signing and verifying has additional overhead.

Reviewer: ```suggestion
It makes sense to sign the value of a cookie if you store important data that should not be modified by a user.
```

- `sameSite`, if set to either `SAME_SITE_LAX` or `SAME_SITE_STRICT` would prevent sending a cookie in cross-site
  browsing context. `SAME_SITE_LAX` would prevent cookie sending during CSRF-prone request methods (e.g. POST, PUT,
  PATCH etc). `SAME_SITE_STRICT` would prevent cookie sending for all methods.

> - Sign the value of the cookie to prevent spoofing of values if the data in the value is important.

Reviewer: ```suggestion
- Sign the value of the cookie to prevent spoofing of values if the data in the value should not be tampered with.
```

> In any case, the use of cookies cannot be 100% secure, so you should not trust the data stored in them and always check this data.

Reviewer: I'd delete the statement. It's very contraversial.
